### ISSUE TITLE
How to use secrets in plugins?

### ISSUE THREAD

**Issue by HerHde** [woodpecker-ci/woodpecker#466]
Hey there,

I'm trying to set up the [Drone Gitea plugin](http://plugins.drone.io/drone-plugins/drone-gitea-release/), *using credentials from secrets*, but I can't figure out how to reference a secret as a value correctly. I tried out setting api_key to `$gitea_token` `$GITEA_TOKEN`, `${GITEA_TOKEN}` and with an attribute `from_secret`, trying to adapt to the different syntax compared to Drone:

```yaml
  release:
    image: plugins/gitea-release
    secrets: [ gitea_token ]
    api_key:
      from_secret: gitea_token
```

Could you help me out on this, please? How do I use secrets for api_key?

**Comment by HerHde** [comment]
May I *push* and ask for anything new?

**Comment by reivilibre** [comment]
Hi, maybe this can help.

Secrets are just environment variables and so are plugin parameters, so with a little bit of trickery you can map them between each other.

Here's an example of the [`docker` plugin](http://plugins.drone.io/drone-plugins/drone-docker/) that I've been using.
```yaml
pipeline:
  dockerImage:
    image: plugins/docker
    dockerfile: "${IMAGE_NAME}/Dockerfile"
    repo: "docker.example.org/me/${IMAGE_NAME}"
    registry: "docker.example.org"
    tags:
      - latest
    secrets:
      - source: docker_registry_username
        target: plugin_username
      - source: docker_registry_password
        target: plugin_password
```

The key is to use `secrets` with the `source` and `target` mappings.

For your example, specify the secret in the web UI as `GITEA_TOKEN` (yes, uppercase!), then use:

```
  release:
    image: plugins/gitea-release
    secrets:
      - source: gitea_token
        target: plugin_api_key
```
(yes, you write them in lowercase in the `secrets` block.)

I noticed this because the [example plugin](https://woodpecker-ci.org/docs/usage/plugins/sample-plugin) shows that plugin parameters are just `PLUGIN_XXX` env vars and the [secrets documentation](https://woodpecker-ci.org/docs/usage/secrets#alternate-names) shows you how to use alternate names.
I don't think this is a very nice way of doing it, but it works today. It might be sensible for a more 'proper' way of doing this to be added to Woodpecker's YAML syntax but I didn't see one so far.

**Comment by HerHde** [comment]
Thank you @reivilibre, this works indeed!

**Comment by markkrj** [comment]
@6543 and @jolheiser I think #604 closes this? Or you wait for 0.15 to be released?

**Comment by 6543** [comment]
yes was closed by #604 - way better than to do it atm :)
